### ISSUE TITLE
Soundfilers no longer save audio correctly

### ISSUE THREAD

**Issue by jamiebullock** [BirminghamConservatoire/IntegraLive#963]
To reproduce:
- Open Integra Live 1.7
- Create any Soundfiler
- Record some audio
- Try to save the recorded audio

Expected:
- An audio file is created with the saved audio

Actual:
- An audio file is created but the audio is not present

**Comment by balandinodidonato** [comment]
Hi Jamie, it is actually working on my machine: Mac OS X Yosemite 10.10.2 (14C1514)

**Comment by jamiebullock** [comment]
@balandinodidonato Are you definitely testing with 1.7? The latest build can be found here: http://integra.io/incoming/Integra%20Live%201.7.0.3305.zip

**Comment by balandinodidonato** [comment]
Yes. To be sure I downloaded it again and it works.

**Comment by jamiebullock** [comment]
Just to be clear: you can definitely play back the audio file created by IL? For me it creates a file, but it is empty.

**Comment by balandinodidonato** [comment]
Hi Jamie, I find out something really weird.

What you say it is actually true, however the sound filer inside the soundfileLoop block from the Block Library works perfectly.

https://www.dropbox.com/s/pdxjt8wz5p3mz5m/soundfiler-test.mov?dl=0

**Comment by jamiebullock** [comment]
Right. That's because the SoundFiler in the block library is using the *old* Soundfiler from 1.6.

Don't forget that when you save a project, track or block, it saves the *actual modules* inside the .integra file. This enables you to always play "old versions" of a project with any future version of Integra, even if we completely change a module or remove it entirely! Users can "upgrade" their projects to use the latest versions, or just keep them as they are, indefinitely.

Interestingly, there is a commit where I upgraded the block library: #944 However, this now seems to have regressed—maybe when I merged some changes between branches. I've therefore reopened the issue.

So we need to do two things: fix the Soundfilers. Update the block library blocks with the new Soundfilers.
